doc/dotNET SDK Proposal.md

```diff
@@ -0,0 +1,232 @@
+
+### Workflow Host Configuration
+
+The user-facing foundation of the SDK is the host. It is configured in a manner that closely follows the conventions for configuring an ASP.NET Core application host ([basic example](../src/Samples/Basic/Temporal.Samples.DesignProcess/Part1_1_TypicalWorkflow.cs#L73-L89) | [more examples](../src/Samples/Basic/Temporal.Samples.DesignProcess/Part4_1_BasicWorkflowUsage.cs#L67-L202) | [another example](../src/Samples/Basic/Temporal.Samples.DesignProcess/Part4_2_BasicWorkflowUsage_MultipleWorkers.cs#L108-L158)).
+
+Users provide workflow implementations and activity implementations in one of two ways:
+ - Pattern based: Entry points of workflow and activity implementations follow specific, well known patterns. Developers use attributes to provide metadata about such entry point to the host. The host automatically loads, and executes workflows and activities. This approach allows writing workflows (and activities) with hardly any scaffolding. 
+- Interface based: Developers implement specific interfaces to create workflows (and activities), and point the host to these implementations. The host automatically loads, and executes workflows and activities. This approach requires a little more scaffolding. However, it also allows additional functionality and helps addressing some advanced scenarios (e.g. dynamic workflow APIs).
+
+### Basic Workflow Execution
+
+Consider a sample workflow of type `SomeWorkflow` that runs a loop executing `SomeActivity` until a certain condition is met. When such condition is eventually met, the workflow is notified via a `NotifyConditionMet` signal.
+
+At application startup, the application is configured to host a worker that can execute workflows (and activities). Then, the worker is configured to poll `SomeTaskQueue` for work items, and to support execution of `SomeWorkflow` instances:
+
+```cs
+public static void Main(string[] args)
+{     
+    IHost appHost = Host.CreateDefaultBuilder(args)
+            .UseTemporalWorkerHost()
+            .ConfigureServices(serviceCollection =>
```

Comment:
Here is an example of providing such config in code:
https://github.com/chronoguy/sdk-dotnet/blob/ffd0b9acaa8645631876200db85505888b4c547d/src/Samples/Basic/Temporal.Samples.DesignProcess/Part4_1_BasicWorkflowUsage.cs#L146-L150

In addition, but using the ASP.NET host, it will automatically use all the built-in ways to read the config from config files.

Reply:
It is set up by the call to `UseTemporalWorkerHost(..)`. Added a clarification to the next doc version.

✔️